src/main/flight/pid.c

```diff
@@ -368,125 +346,87 @@ float pidApplyThrustLinearization(float motorOutput)
 #endif
 
 #if defined(USE_ACC)
-// calculate the stick deflection while applying level mode expo
-static float getAngleModeStickInputRaw(uint8_t axis)
-{
-    const float stickDeflection = getRcDeflectionRaw(axis);
-    if (axis < FD_YAW) {
-        const float expof = currentControlRateProfile->levelExpo[axis] / 100.0f;
-        return power3(stickDeflection) * expof + stickDeflection * (1 - expof);
-    } else {
-        return stickDeflection;
-    }
-}
-
-// calculates strength of horizon leveling; 0 = none, 1.0 = most leveling
+// Calculate strength of horizon leveling; 0 = none, 1.0 = most leveling
 STATIC_UNIT_TESTED FAST_CODE_NOINLINE float calcHorizonLevelStrength(void)
 {
-    const float currentInclination = MAX(abs(attitude.values.roll), abs(attitude.values.pitch)) / 10.0f;
+    const float currentInclination = MAX(abs(attitude.values.roll), abs(attitude.values.pitch)) * 0.1f;
     // 0 when level, 90 when vertical, 180 when inverted (degrees):
-    float horizonLevelStrength = MAX((pidRuntime.horizonLimitDegrees - currentInclination) / pidRuntime.horizonLimitDegrees, 0.0f);
+
+    const float hlim = pidRuntime.horizonLimitDegrees;
+    const float hlimInv = pidRuntime.horizonLimitDegreesInv;
+    float horizonLevelStrength = MAX((hlim - currentInclination) * hlimInv, 0.0f);
     // 1.0 when attitude is 'flat', 0 when angle is equal to, or greater than, horizonLimitDegrees
 
     if (!pidRuntime.horizonIgnoreSticks) {
-    // horizonIgnoreSticks:  0 = default; levelling attenuated by both attitude and sticks;
-    //                       1 = level attenuation only by attitude
-        const float absMaxStickDeflection = MAX(fabsf(getRcDeflection(FD_ROLL)), fabsf(getRcDeflection(FD_PITCH))); // 0-1, smoothed if RC smoothing is enabled
-        const float horizonStickAttenuation = MAX((pidRuntime.horizonLimitSticks - absMaxStickDeflection) / pidRuntime.horizonLimitSticks, 0.0f);
+        // horizonIgnoreSticks:  0 = default; levelling attenuated by both attitude and sticks;
+        //                       1 = level attenuation only by attitude
+        const float hLimSticks = pidRuntime.horizonLimitSticks;
+        const float hLimSticksInv = pidRuntime.horizonLimitSticksInv;
+        const float horizGain = pidRuntime.horizonGain;
```

Comment:
Not needed
```suggestion
```

Reply:
You're right, makes no difference.  Removed these declarations and optimised the code significantly.